daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SpecifiedLengthParsers.scala

```diff
@@ -132,6 +133,20 @@ class SpecifiedLengthExplicitParser(
   }
 }
 
+class SpecifiedLengthChoiceParser(
+  eParser: Parser,
+  erd: RuntimeData,
+  lengthEv: Evaluatable[JLong],
+  toBits: Int)
+  extends SpecifiedLengthParserBase(eParser, erd) {
+
+  final override def getBitLength(s: PState): MaybeULong = {
+    val nBytesAsAny = lengthEv.evaluate(s)
```

Comment:
suggest name choiceLengthEv. There is already a lengthEv, so this is confusing.

Reply:
Oh, yeah and you don't need this beause it's not an expression. But call it choiceLength anyway.

Reply:
I've removed all Ev stuff and just reimpemented a simplified version of ElementUnusedUnparser without evaluatables.